sources/polyfills/MapLike.ts

```diff
@@ -0,0 +1,64 @@
+declare global {
+	class Map<K, V> {
+		// es2015.collection.d.ts
+		clear(): void;
+		delete(key: K): boolean;
+		forEach(
+			callbackfn: (value: V, key: K, map: Map<K, V>) => void,
+			thisArg?: any,
+		): void;
+		get(key: K): V | undefined;
+		has(key: K): boolean;
+		set(key: K, value: V): this;
+		readonly size: number;
+	}
+}
+
+class MapLike<K, V> {
+	private keys: K[];
+	private values: V[];
```

Comment:
i'm not sure if this is the best way to do this, but i couldn't think of another way that would let us use objects as "keys" in the map since normal object keys can't be other objects